### PR TITLE
dev/reference: add some description about max_execution_time

### PR DESCRIPTION
**Pull request by tiancaiamao**
### What is changed, added or deleted? 

Tiny update the document.

### What is the related PR or file link(s)? 

https://github.com/pingcap/docs-cn/pull/1721

### Which version does your change affect? 

dev

### Checklist 

- [ ] Add a new file to `TOC.md`
- [ ] No Tab spaces anywhere 
- [ ] Leave a blank line both before and after a code block
- [ ] Keep the first level heading consistent with `title` in metadata
- [ ] Use *four* spaces for each level of indentation except that in `TOC.md`

### CODE REVIEW

**Comment by tiancaiamao** [comment]
PTAL @jackysp @dcalvin

**Comment by tiancaiamao** [comment]
What's the problem with the lint check... I didn't modify those parts!

**Comment by yikeke** [comment]
> What's the problem with the lint check... I didn't modify those parts!

I will fix those parts for you.

**Comment by yikeke** [comment]
> > What's the problem with the lint check... I didn't modify those parts!
> 
> Those are the remaining lint issues in our docs. I will fix those parts for you.

Actually those issues have been fixed in the upstream master, so I just merged 'master' into your branch and the check has passed!

**Comment by lilin90** [comment]
@tiancaiamao Does your change only affect `dev` which means `v4.0`?

File: dev/reference/configuration/tidb-server/mysql-variables.md [hunk 1]
```diff
@@ -110,6 +110,9 @@ The following MySQL system variables are fully supported in TiDB and have the sa
 | time_zone | GLOBAL \| SESSION | the time zone of the database |
 | tx_isolation | GLOBAL \| SESSION | the isolation level of a transaction |
 | hostname | NONE | the hostname of the TiDB server |
+| max\_execution\_time | GLOBAL \| SESSION |  the execution timeout for a statement, in milliseconds |
+
+Note, unlike MySQL, the `max_execution_time` system variable currently works on all kinds of statements in TiDB, it's not only restricted to the `select` statement. The precision of the timeout value is roughly 100ms, that means the statement may not be terminated in accurate milliseconds as you specified.
```
Comment (dcalvin) [inline dev/reference/configuration/tidb-server/mysql-variables.md]:
```suggestion
Note that, unlike MySQL,  the `max_execution_time` system variable currently works on all kinds of statements in TiDB, and it's not only restricted to the `SELECT` statement. The precision of the timeout value is roughly 100ms. This means the statement might not be terminated in accurate milliseconds as you specified.
```

File: dev/reference/performance/optimizer-hints.md [hunk 2]
```diff
@@ -35,3 +35,19 @@ SELECT /*+ TIDB_HJ(t1, t2) */ * from t1, t2 where t1.id = t2.id
...
+MAX_EXECUTION_TIME(N)
+```
+
+Example with a timeout of 1 second (1000 milliseconds):
+
+```sql
+SELECT /*+ MAX_EXECUTION_TIME(1000) */ * FROM t1 INNER JOIN t2 WHERE ...
+```
+
+In addition to the hint, the `max_execution_time` system variable also applies to the current statement.
```
Comment (dcalvin) [inline dev/reference/performance/optimizer-hints.md]:
```suggestion
In addition to this hint, the `max_execution_time` system variable also limits the execution time of a statement.
```

**Comment by tiancaiamao** [comment]
This change affects v2.1 v3.0 and master, I've updated the document @yikeke

**Comment by yikeke** [comment]
To resolve the remaining markdown lint issues: see the MD001 rule in https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md. @tiancaiamao

**Comment by tiancaiamao** [comment]
Fixed @yikeke

File: dev/reference/configuration/tidb-server/mysql-variables.md [hunk 3]
```diff
@@ -110,6 +110,9 @@ The following MySQL system variables are fully supported in TiDB and have the sa
 | time_zone | GLOBAL \| SESSION | the time zone of the database |
 | tx_isolation | GLOBAL \| SESSION | the isolation level of a transaction |
 | hostname | NONE | the hostname of the TiDB server |
+| max\_execution\_time | GLOBAL \| SESSION |  the execution timeout for a statement, in milliseconds |
+
+Note that, unlike MySQL, the `max_execution_time` system variable currently works on all kinds of statements in TiDB, and it's not only restricted to the `SELECT` statement. The precision of the timeout value is roughly 100ms. This means the statement might not be terminated in accurate milliseconds as you specified.
```
Comment (lilin90) [inline dev/reference/configuration/tidb-server/mysql-variables.md]:
```suggestion
> **Note:**
>
> Unlike in MySQL, the `max_execution_time` system variable currently works on all kinds of statements in TiDB, not only restricted to the `SELECT` statement. The precision of the timeout value is roughly 100ms. This means the statement might not be terminated in accurate milliseconds as you specify.
```

File: v2.1/reference/configuration/tidb-server/mysql-variables.md [hunk 4]
```diff
@@ -110,6 +110,9 @@ The following MySQL system variables are fully supported in TiDB and have the sa
 | time_zone | GLOBAL \| SESSION | the time zone of the database |
 | tx_isolation | GLOBAL \| SESSION | the isolation level of a transaction |
 | hostname | NONE | the hostname of the TiDB server |
+| max\_execution\_time | GLOBAL \| SESSION |  the execution timeout for a statement, in milliseconds |
+
+ Note that, unlike MySQL, the `max_execution_time` system variable currently works on all kinds of statements in TiDB, and it's not only restricted to the `SELECT` statement. The precision of the timeout value is roughly 100ms. This means the statement might not be terminated in accurate milliseconds as you specified.
```
Comment (lilin90) [inline v2.1/reference/configuration/tidb-server/mysql-variables.md]:
```suggestion
> **Note:**
>
> Unlike in MySQL, the `max_execution_time` system variable currently works on all kinds of statements in TiDB, not only restricted to the `SELECT` statement. The precision of the timeout value is roughly 100ms. This means the statement might not be terminated in accurate milliseconds as you specify.
```

File: v3.0/reference/configuration/tidb-server/mysql-variables.md [hunk 5]
```diff
@@ -111,6 +111,9 @@ The following MySQL system variables are fully supported in TiDB and have the sa
 | time_zone | GLOBAL \| SESSION | the time zone of the database |
 | tx_isolation | GLOBAL \| SESSION | the isolation level of a transaction |
 | hostname | NONE | the hostname of the TiDB server |
+| max\_execution\_time | GLOBAL \| SESSION |  the execution timeout for a statement, in milliseconds |
+
+ Note that, unlike MySQL, the `max_execution_time` system variable currently works on all kinds of statements in TiDB, and it's not only restricted to the `SELECT` statement. The precision of the timeout value is roughly 100ms. This means the statement might not be terminated in accurate milliseconds as you specified.
```
Comment (lilin90) [inline v3.0/reference/configuration/tidb-server/mysql-variables.md]:
```suggestion
> **Note:**
>
> Unlike in MySQL, the `max_execution_time` system variable currently works on all kinds of statements in TiDB, not only restricted to the `SELECT` statement. The precision of the timeout value is roughly 100ms. This means the statement might not be terminated in accurate milliseconds as you specify.
```

File: v2.1/reference/performance/optimizer-hints.md [hunk 6]
```diff
@@ -12,26 +12,44 @@ TiDB supports optimizer hints, based on the comment-like syntax introduced in My
...
 
 ```sql
 SELECT /*+ TIDB_HJ(t1, t2) */ * from t1, t2 where t1.id = t2.id
 ```
 
 This variable is used to remind the optimizer to use the `Hash Join` algorithm. This algorithm executes threads concurrently. It runs faster but takes up more memory.
+
+## MAX\_EXECUTION\_TIME(N)
+
+ The `MAX_EXECUTION_TIME` hint is permitted only for SELECT statements. It places a limit N (a timeout value in milliseconds) on how long a statement is permitted to execute before the server terminates it:
```
Comment (lilin90) [inline v2.1/reference/performance/optimizer-hints.md]:
```suggestion
The `MAX_EXECUTION_TIME` hint is permitted only for SELECT statements. It places a limit N (a timeout value in milliseconds) on how long a statement is permitted to execute before the server terminates it:
```

File: v2.1/reference/performance/optimizer-hints.md [hunk 7]
```diff
@@ -12,26 +12,44 @@ TiDB supports optimizer hints, based on the comment-like syntax introduced in My
...
+
+ ```
```
Comment (lilin90) [inline v2.1/reference/performance/optimizer-hints.md]:
```suggestion
```
```

File: v2.1/reference/performance/optimizer-hints.md [hunk 8]
```diff
@@ -12,26 +12,44 @@ TiDB supports optimizer hints, based on the comment-like syntax introduced in My
...
+MAX_EXECUTION_TIME(N)
+```
+
+ Example with a timeout of 1 second (1000 milliseconds):
```
Comment (lilin90) [inline v2.1/reference/performance/optimizer-hints.md]:
```suggestion
Example with a timeout of 1 second (1000 milliseconds):
```

File: v2.1/reference/performance/optimizer-hints.md [hunk 9]
```diff
@@ -12,26 +12,44 @@ TiDB supports optimizer hints, based on the comment-like syntax introduced in My
...
+
+ ```sql
```
Comment (lilin90) [inline v2.1/reference/performance/optimizer-hints.md]:
```suggestion
```sql
```

File: v2.1/reference/performance/optimizer-hints.md [hunk 10]
```diff
@@ -12,26 +12,44 @@ TiDB supports optimizer hints, based on the comment-like syntax introduced in My
...
+SELECT /*+ MAX_EXECUTION_TIME(1000) */ * FROM t1 INNER JOIN t2 WHERE ...
+```
+
+ In addition to this hint, the `max_execution_time` system variable also limits the execution time of a statement.
```
Comment (lilin90) [inline v2.1/reference/performance/optimizer-hints.md]:
```suggestion
In addition to this hint, the `max_execution_time` system variable also limits the execution time of a statement.
```

File: v3.0/reference/performance/optimizer-hints.md [hunk 11]
```diff
@@ -13,26 +13,42 @@ TiDB supports optimizer hints, based on the comment-like syntax introduced in My
...
 
 ```sql
 SELECT /*+ TIDB_HJ(t1, t2) */ * from t1, t2 where t1.id = t2.id
 ```
 
 This variable is used to remind the optimizer to use the `Hash Join` algorithm. This algorithm executes threads concurrently. It runs faster but takes up more memory.
+
+## MAX\_EXECUTION\_TIME(N)
+
+ The `MAX_EXECUTION_TIME` hint is permitted only for SELECT statements. It places a limit N (a timeout value in milliseconds) on how long a statement is permitted to execute before the server terminates it:
```
Comment (lilin90) [inline v3.0/reference/performance/optimizer-hints.md]:
```suggestion
The `MAX_EXECUTION_TIME` hint is permitted only for SELECT statements. It places a limit N (a timeout value in milliseconds) on how long a statement is permitted to execute before the server terminates it:
```

File: v3.0/reference/performance/optimizer-hints.md [hunk 12]
```diff
@@ -13,26 +13,42 @@ TiDB supports optimizer hints, based on the comment-like syntax introduced in My
...
+
+ ```
```
Comment (lilin90) [inline v3.0/reference/performance/optimizer-hints.md]:
```suggestion
```
```

File: v3.0/reference/performance/optimizer-hints.md [hunk 13]
```diff
@@ -13,26 +13,42 @@ TiDB supports optimizer hints, based on the comment-like syntax introduced in My
...
+MAX_EXECUTION_TIME(N)
+```
+
+ Example with a timeout of 1 second (1000 milliseconds):
```
Comment (lilin90) [inline v3.0/reference/performance/optimizer-hints.md]:
```suggestion
Example with a timeout of 1 second (1000 milliseconds):
```

File: v3.0/reference/performance/optimizer-hints.md [hunk 14]
```diff
@@ -13,26 +13,42 @@ TiDB supports optimizer hints, based on the comment-like syntax introduced in My
...
+
+ ```sql
```
Comment (lilin90) [inline v3.0/reference/performance/optimizer-hints.md]:
```suggestion
```sql
```

File: v3.0/reference/performance/optimizer-hints.md [hunk 15]
```diff
@@ -13,26 +13,42 @@ TiDB supports optimizer hints, based on the comment-like syntax introduced in My
...
+SELECT /*+ MAX_EXECUTION_TIME(1000) */ * FROM t1 INNER JOIN t2 WHERE ...
+```
+
+ In addition to this hint, the `max_execution_time` system variable also limits the execution time of a statement.
```
Comment (lilin90) [inline v3.0/reference/performance/optimizer-hints.md]:
```suggestion
In addition to this hint, the `max_execution_time` system variable also limits the execution time of a statement.
```

**Comment by tiancaiamao** [comment]
I don't know why there is a white space at the beginning of each line, I don't mean to, that's weird.

**Comment by tiancaiamao** [comment]
Fixed @lilin90 @yikeke

File: dev/reference/performance/optimizer-hints.md [hunk 16]
```diff
@@ -35,3 +35,19 @@ SELECT /*+ TIDB_HJ(t1, t2) */ * from t1, t2 where t1.id = t2.id
 ```
 
 This variable is used to remind the optimizer to use the `Hash Join` algorithm. This algorithm executes threads concurrently. It runs faster but takes up more memory.
+
+### MAX\_EXECUTION\_TIME(N)
+
+The `MAX_EXECUTION_TIME` hint is permitted only for SELECT statements. It places a limit N (a timeout value in milliseconds) on how long a statement is permitted to execute before the server terminates it:
```
Comment (lilin90) [inline dev/reference/performance/optimizer-hints.md]: Would it be better this way?

```suggestion
The `MAX_EXECUTION_TIME` hint only applies to `SELECT` statements. It places a limit `N` (a timeout value in milliseconds) on how long a statement is permitted to execute before the server terminates it:
```

File: v2.1/reference/performance/optimizer-hints.md [hunk 17]
```diff
@@ -12,26 +12,42 @@ TiDB supports optimizer hints, based on the comment-like syntax introduced in My
...
 
 ```sql
 SELECT /*+ TIDB_HJ(t1, t2) */ * from t1, t2 where t1.id = t2.id
 ```
 
 This variable is used to remind the optimizer to use the `Hash Join` algorithm. This algorithm executes threads concurrently. It runs faster but takes up more memory.
+
+## MAX\_EXECUTION\_TIME(N)
+
+The `MAX_EXECUTION_TIME` hint is permitted only for SELECT statements. It places a limit N (a timeout value in milliseconds) on how long a statement is permitted to execute before the server terminates it:
```
Comment (lilin90) [inline v2.1/reference/performance/optimizer-hints.md]:
```suggestion
The `MAX_EXECUTION_TIME` hint only applies to `SELECT` statements. It places a limit `N` (a timeout value in milliseconds) on how long a statement is permitted to execute before the server terminates it:
```

File: v3.0/reference/performance/optimizer-hints.md [hunk 18]
```diff
@@ -13,26 +13,40 @@ TiDB supports optimizer hints, based on the comment-like syntax introduced in My
...
 
 ```sql
 SELECT /*+ TIDB_HJ(t1, t2) */ * from t1, t2 where t1.id = t2.id
 ```
 
 This variable is used to remind the optimizer to use the `Hash Join` algorithm. This algorithm executes threads concurrently. It runs faster but takes up more memory.
+
+## MAX\_EXECUTION\_TIME(N)
+
+The `MAX_EXECUTION_TIME` hint is permitted only for SELECT statements. It places a limit N (a timeout value in milliseconds) on how long a statement is permitted to execute before the server terminates it:
```
Comment (lilin90) [inline v3.0/reference/performance/optimizer-hints.md]:
```suggestion
The `MAX_EXECUTION_TIME` hint only applies to `SELECT` statements. It places a limit `N` (a timeout value in milliseconds) on how long a statement is permitted to execute before the server terminates it:
```

**Comment by lilin90** [comment]
@jackysp PTAL

**Comment by jackysp** [comment]
LGTM

File: v2.1/reference/performance/optimizer-hints.md [hunk 19]
```diff
@@ -12,26 +12,42 @@ TiDB supports optimizer hints, based on the comment-like syntax introduced in My
 >
 > MySQL command-line clients earlier than 5.7.7 strip optimizer hints by default. If you want to use the `Hint` syntax in these earlier versions, add the `--comments` option when starting the client. For example: `mysql -h 127.0.0.1 -P 4000 -uroot --comments`.
 
-### TIDB_SMJ(t1, t2)
+## Optimizer Hint Syntax
+
+## TIDB_SMJ(t1, t2)
```
Comment (jackysp) [inline v2.1/reference/performance/optimizer-hints.md]:
```suggestion
### TIDB_SMJ(t1, t2)
```

File: v2.1/reference/performance/optimizer-hints.md [hunk 20]
```diff
@@ -12,26 +12,42 @@ TiDB supports optimizer hints, based on the comment-like syntax introduced in My
...
 
 ```sql
 SELECT /*+ TIDB_SMJ(t1, t2) */ * from t1, t2 where t1.id = t2.id
 ```
 
 This variable is used to remind the optimizer to use the `Sort Merge Join` algorithm. This algorithm takes up less memory, but takes longer to execute. It is recommended if the data size is too large, or there’s insufficient system memory.
 
-### TIDB_INLJ(t1, t2)
+## TIDB_INLJ(t1, t2)
```
Comment (jackysp) [inline v2.1/reference/performance/optimizer-hints.md]:
```suggestion
### TIDB_INLJ(t1, t2)
```

File: v2.1/reference/performance/optimizer-hints.md [hunk 21]
```diff
@@ -12,26 +12,42 @@ TiDB supports optimizer hints, based on the comment-like syntax introduced in My
...
 
 ```sql
 SELECT /*+ TIDB_INLJ(t1, t2) */ * from t1, t2 where t1.id = t2.id
 ```
 
 This variable is used to remind the optimizer to use the `Index Nested Loop Join` algorithm. In some scenarios, this algorithm runs faster and takes up fewer system resources, but may be slower and takes up more system resources in some other scenarios. You can try to use this algorithm in scenarios where the result-set is less than 10,000 rows after the outer table is filtered by the WHERE condition. The parameter in `TIDB_INLJ()` is the candidate table for the inner table when you create the query plan. For example, `TIDB_INLJ (t1)` means that TiDB only considers using t1 as the inner table to create a query plan.
 
-### TIDB_HJ(t1, t2)
+## TIDB_HJ(t1, t2)
```
Comment (jackysp) [inline v2.1/reference/performance/optimizer-hints.md]:
```suggestion
### TIDB_HJ(t1, t2)
```

File: v2.1/reference/performance/optimizer-hints.md [hunk 22]
```diff
@@ -12,26 +12,42 @@ TiDB supports optimizer hints, based on the comment-like syntax introduced in My
...
 
 ```sql
 SELECT /*+ TIDB_HJ(t1, t2) */ * from t1, t2 where t1.id = t2.id
 ```
 
 This variable is used to remind the optimizer to use the `Hash Join` algorithm. This algorithm executes threads concurrently. It runs faster but takes up more memory.
+
+## MAX\_EXECUTION\_TIME(N)
```
Comment (jackysp) [inline v2.1/reference/performance/optimizer-hints.md]:
```suggestion
### MAX\_EXECUTION\_TIME(N)
```

File: v3.0/reference/performance/optimizer-hints.md [hunk 23]
```diff
@@ -13,26 +13,40 @@ TiDB supports optimizer hints, based on the comment-like syntax introduced in My
 >
 > MySQL command-line clients earlier than 5.7.7 strip optimizer hints by default. If you want to use the `Hint` syntax in these earlier versions, add the `--comments` option when starting the client. For example: `mysql -h 127.0.0.1 -P 4000 -uroot --comments`.
 
-### TIDB_SMJ(t1, t2)
+## TIDB_SMJ(t1, t2)
```
Comment (jackysp) [inline v3.0/reference/performance/optimizer-hints.md]: Please use the same format as v2.1.

File: dev/reference/performance/optimizer-hints.md [hunk 24]
```diff
@@ -35,3 +35,19 @@ SELECT /*+ TIDB_HJ(t1, t2) */ * from t1, t2 where t1.id = t2.id
 ```
 
 This variable is used to remind the optimizer to use the `Hash Join` algorithm. This algorithm executes threads concurrently. It runs faster but takes up more memory.
+
+### MAX\_EXECUTION\_TIME(N)
```
Comment (jackysp) [inline dev/reference/performance/optimizer-hints.md]: Please use the same format as v2.1.

**Comment by lilin90** [comment]
The body headings are not the same as docs-cn (https://github.com/pingcap/docs-cn/pull/1721) but it's also acceptable.

**Pull request merged**